hy/core/macros.hy

```diff
@@ -20,6 +20,16 @@
          ~@(interleave (repeat name) rest))
      ~name))
 
+
+(defmacro assoc [coll k1 v1 &rest other-kvs]
+  (if (odd? (len other-kvs))
+    (macro-error (last other-kvs) "`assoc` takes an odd number of arguments"))
+  (setv c (if other-kvs (gensym "c") coll))
+  `(setv ~@(+ (if other-kvs [c coll] []) #* (genexpr
+    [`(get ~c ~k) v]
+    [[k v] (partition (+ (, k1 v1) other-kvs))]))))
```

Comment:
No, it is not indented 2 spaces. After bracket counting, I can see that arguments to `genexpr` are indented as if they were arguments to `setv`. It will take 42 more spaces to get it indented two spaces past the start of their list. To avoid excessively wide lines, instead of just inserting 42 spaces at the start of those two lines, add some newlines.

One must never put list elements on a column before (or equal to) the bracket opening the list, in any Lisp. Elements must always appear on a column following the opening bracket of the list they belong to. And when I say "list", this applies to all Hy models inheriting from HyList.